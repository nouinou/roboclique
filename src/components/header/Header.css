.header {
    background-color: #3E4451;
    min-height: 20vh;
    width: 100%;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    font-size: calc(10px + 2vmin);
    color: white;
    text-transform: uppercase;
}
  