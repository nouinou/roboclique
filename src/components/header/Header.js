const Header = () => {
	return (
		<div className="header flex flex-col items-center justify-center text-white robo-text w-screen min-h-20 bg-primary p-5 mb-6">
			<h1 className="robo-text text-4xl md:text-7xl">Robo Clique</h1>
		</div>
	);
};

export default Header;
