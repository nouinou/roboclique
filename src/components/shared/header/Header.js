import { PureComponent } from 'react';
import SearchBox from '../search-box/SearchBox';

class Header extends PureComponent {
	render() {
		return (
			<div className="header flex flex-col items-center justify-center w-screen min-h-20 bg-primary p-5 mb-6">
				<h1 className="robo-text text-5xl text-white md:text-7xl">
					Robo Clique
				</h1>
				<SearchBox searchChange={this.props.searchTermChanged} />
			</div>
		);
	}
}

export default Header;
