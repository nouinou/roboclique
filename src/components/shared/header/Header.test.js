import { shallow, mount } from 'enzyme';
import Header from './Header';

describe('Header Component', () => {
	const mockSearchTermChanged = jest.fn();

	it('should render Header', () => {
		expect(
			shallow(<Header searchTermChanged={mockSearchTermChanged} />)
		).toMatchSnapshot();
	});

	it('should call searchTermChanged', () => {
		const wrapper = mount(<Header searchTermChanged={mockSearchTermChanged} />);
		const spy = jest.spyOn(wrapper.instance().props, 'searchTermChanged');
		wrapper
			.find('input')
			.at(0)
			.simulate('change', { target: { value: 'robot' } });
		expect(spy).toBeCalled();
	});
});
