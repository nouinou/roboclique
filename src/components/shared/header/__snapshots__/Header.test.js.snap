// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should render Header 1`] = `
<div
  className="header flex flex-col items-center justify-center w-screen min-h-20 bg-primary p-5 mb-6"
>
  <h1
    className="robo-text text-5xl text-white md:text-7xl"
  >
    Robo Clique
  </h1>
  <SearchBox
    searchChange={[MockFunction]}
  />
</div>
`;
